app/samples/src/main/java/com/sygic/samples/app/dialogs/AboutDialog.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019 Sygic a.s. All rights reserved.
+ *
+ * This project is licensed under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.samples.app.dialogs
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.lifecycle.ViewModelProviders
+import com.sygic.samples.databinding.LayoutAboutDialogBinding
+import com.sygic.samples.app.viewmodels.AboutDialogViewModel
+
+class AboutDialog : AppCompatDialogFragment() {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val binding = LayoutAboutDialogBinding.inflate(inflater, container, false)
+        binding.aboutDialogViewModel = ViewModelProviders.of(
+            this,
+            AboutDialogViewModel.ViewModelFactory(requireActivity().application)
```

Comment:
ak je ten VM `AndroidViewModel` a jeho jediny parameter je `App`, podla mna factorku netreba. Mal by to automaticky vytvorit. Tak ako nepotrebujes pre `ViewModel`, ktory nema parametre.

Reply:
teda predpokladam, ze to tak urobili ;)